samples/bluetooth/peripheral_hr/Kconfig.sysbuild

```diff
@@ -13,3 +13,28 @@ config NET_CORE_IMAGE_HCI_IPC
 	bool "HCI IPC image on network core"
 	default y
 	depends on NET_CORE_BOARD != ""
+
+config BT_CTLR_SCAN_DATA_LEN_MAX
+	int "Scan data length max"
+	default 36
+	depends on TEST_MODE_EXTENDED || TEST_MODE_PHY_CODED
+
+config BT_BUF_EVT_DISCARDABLE_SIZE
+	int "Discardable buffer size"
+	default 58
+	depends on TEST_MODE_EXTENDED || TEST_MODE_PHY_CODED
+
+choice TEST_MODE
+	prompt "Test mode"
+	default TEST_MODE_MINIMAL
+
+config TEST_MODE_MINIMAL
+	bool "Minimal"
+
+config TEST_MODE_EXTENDED
+	bool "Extended"
+
+config TEST_MODE_PHY_CODED
+	bool "PHY coded"
```

Comment:
Why aren't these defined in the BSIM test application instead?

Reply:
They're used by the normal application, e.g. configure for `nrf5340dk//cpuapp` using sysbuild with `-DSB_CONFIG_TEST_MODE_EXTENDED=y` and you get `BT_BUF_EVT_DISCARDABLE_SIZE` propagated to both images:
```
zephyr/.config:SB_CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE=58
peripheral_hr/zephyr/.config.sysbuild:CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE=58
peripheral_hr/zephyr/.config:CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE=58
hci_ipc/zephyr/.config.sysbuild:CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE=58
hci_ipc/zephyr/.config:CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE=58
```

Reply:
Aren't `TEST` and `normal application` somewhat mutually exclusive?

Reply:
I have no clue on that, I just updated the previous PR to use sysbuild as presented here. @alwa-nordic @cvinayak

Reply:
The goal was to cover the samples as-is (atleast, but with some minimal verification) in bsim CI. 

`TEST_MODE` --> `SAMPLE_MODE` ? is that a good name to convey the application's use of advertising extensions?